libraries/protocol/include/graphene/protocol/asset_ops.hpp

```diff
@@ -446,10 +446,20 @@ namespace graphene { namespace protocol {
          uint64_t fee = 20 * GRAPHENE_BLOCKCHAIN_PRECISION;
       };
 
+      struct additional_options_type
+      {
+         /// Which asset to claim fees from. This is needed, e.g., to claim collateral-
+         /// denominated fees from a collateral-backed smart asset. If unset, assumed to be same
+         /// asset as amount_to_claim is denominated in, such as would be the case when claiming
+         /// market fees.
+         fc::optional<asset_id_type> claim_from_asset_id;
```

Comment:
Perhaps add a check in `validate()` that requires `claim_from_asset_id != amount_to_claim.asset_id`?

Reply:
Had thought about this.  Do we in general prefer not allowing "more than one way to make the same request"?  I wonder if there is some utility in leaving the format flexible, i.e. to simplify the request format for a bot making periodic claims on fees of both types.

My gut feels more comfortable putting the check in, as you suggest.  But not able to fully put my finger on why.  Will add the check.

Reply:
Such a check keeps chain size smaller by avoiding unnecessary redundant data.

Reply:
Added in 02f9da8.